gcp/src/main/java/org/apache/iceberg/gcp/GCPProperties.java

```diff
@@ -90,6 +95,10 @@ public GCPProperties(Map<String, String> properties) {
       gcsOAuth2TokenExpiresAt =
           new Date(Long.parseLong(properties.get(GCS_OAUTH2_TOKEN_EXPIRES_AT)));
     }
+    gcsNoAuth = Boolean.parseBoolean(properties.getOrDefault(GCS_NO_AUTH, "false"));
+    Preconditions.checkState(
```

Comment:
it would be good to also have a test that specifically checks that you can only set one of these